extensions/replication/tasks/CopyLocationTask.js

```diff
@@ -383,20 +412,38 @@ class CopyLocationTask extends BackbeatTask {
         }, actionEntry.getLogInfo()));
         // A 0-byte object has no range, otherwise range is inclusive.
         const size = range ? range.end - range.start + 1 : 0;
-        let sourceReq = null;
+        const abortController = new AbortController();
+        
         const { bucket, key, version } = actionEntry.getAttribute('target');
         if (size !== 0) {
-            sourceReq = this.backbeatClient.getObject({
+            const getObjectCommand = new GetObjectCommand({
                 Bucket: bucket,
                 Key: key,
                 VersionId: version,
-                // A 0-byte part has no range.
                 Range: range && `bytes=${range.start}-${range.end}`,
                 LocationConstraint: objMD.getDataStoreName(),
+                RequestUids: log.getSerializedUids(),
             });
+            
+            return this.backbeatClient.send(getObjectCommand, { abortSignal: abortController.signal })
+                .then(response => this._putMPUPart(actionEntry, objMD, response.Body, size,
+                        uploadId, partNumber, log, abortController, done))
+                .catch(err => {
+                    if (err.$metadata?.httpStatusCode === 404) {
+                        return done(err);
+                    }
+                    log.error('an error occurred on getObject from S3',
+                    Object.assign({
+                        method: 'CopyLocationTask._getRangeAndPutMPUPartOnce',
+                        error: err.message,
+                        httpStatus: err.$metadata?.httpStatusCode,
+                    }, actionEntry.getLogInfo()));
+                    return done(err);
+                });
         }
-        return this._putMPUPart(actionEntry, objMD, sourceReq, size,
-            uploadId, partNumber, log, done);
+
+        return this._putMPUPart(actionEntry, objMD, undefined, size,
+            uploadId, partNumber, log, abortController, done);
```

Comment:
maybe better to keep the flow...

kind of lends itself better to async flow, but need to catch exception, so wrapping with
`utils.callbackify`:

```suggestion
utils.callbackify(async () => {
    if (!size) {
        return undefined;
    }

    const getObjectCommand = new GetObjectCommand({
        Bucket: bucket,
        Key: key,
        VersionId: version,
        Range: range && `bytes=${range.start}-${range.end}`,
        LocationConstraint: objMD.getDataStoreName(),
        RequestUids: log.getSerializedUids(),
    });
    const resp := await this.backbeatClient.send(getObjectCommand, { abortSignal: abortController.signal });
    return resp.Body;
}), (err, body) => 
    if (err) {
        ...
    } else {
        this._putMPUPart(actionEntry, objMD, body, size, uploadId, partNumber, log, abortController, done));
    }
);
```